javascript/ql/src/semmle/javascript/DefUse.qll

```diff
@@ -182,14 +182,26 @@ class VarDef extends ControlFlowNode {
   /** Gets a variable defined by this node, if any. */
   Variable getAVariable() { result = getTarget().(BindingPattern).getAVariable() }
 
+  private AST::ValueNode getTargetSource(Expr target) { defn(this, target, result) }
+
   /**
    * Gets the source of this definition, that is, the data flow node representing
    * the value that this definition assigns to its target.
    *
    * This predicate is not defined for `VarDef`s where the source is implicit,
    * such as `for-in` loops or parameters.
    */
-  AST::ValueNode getSource() { defn(this, _, result) }
+  AST::ValueNode getSource() {
+    exists(Expr target |
+      not target instanceof DestructuringPattern and getTargetSource(target) = result
+    )
+  }
+
+  /**
+   * Gets the source that this definition destructs, that is, the data flow node representing
+   * the value that contains a value that this definition assigns to its target.
```

Comment:
This is very hard to understand. Could you please rephrase?

Also, I don't see anything in this predicate that restricts it to the destructuring case. It seems to be a strict superset of `getSource`.